File: backend/trips-service/internal/modules/trip/domain/trip.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type TripStatus string

const (
	TripDraft      TripStatus = "draft"
	TripPlanned    TripStatus = "planned"
	TripInProgress TripStatus = "in_progress"
	TripCompleted  TripStatus = "completed"
	TripArchived   TripStatus = "archived"
)

type Trip struct {
	ID           uuid.UUID  `json:"id"`
	OwnerID      uuid.UUID  `json:"owner_id"`
	Title        string     `json:"title"`
	Subtitle     *string    `json:"subtitle"`
	StartDate    *time.Time `json:"start_date"`
	EndDate      *time.Time `json:"end_date"`
	CoverMediaID *uuid.UUID `json:"cover_media_id"`
	Status       TripStatus `json:"status"`
	Region       *string    `json:"region"`
	CenterLat    *float64   `json:"center_lat"`
	CenterLng    *float64   `json:"center_lng"`
	StopsCount   int        `json:"stops_count"`
	EntriesCount int        `json:"entries_count"`
	PhotosCount  int        `json:"photos_count"`
	CreatedAt    time.Time  `json:"created_at"`
	UpdatedAt    time.Time  `json:"updated_at"`
}

type TripStop struct {
	ID        uuid.UUID  `json:"id"`
	TripID    uuid.UUID  `json:"trip_id"`
	PlaceID   *uuid.UUID `json:"place_id"`
	Title     string     `json:"title"`
	Latitude  float64    `json:"latitude"`
	Longitude float64    `json:"longitude"`
	PlannedAt *time.Time `json:"planned_at"`
	VisitedAt *time.Time `json:"visited_at"`
	Notes     *string    `json:"notes"`
	SortOrder int        `json:"sort_order"`
	CreatedAt time.Time  `json:"created_at"`
}

File: landmark_app/lib/features/trips/domain/entities/trip.dart
class Trip {
  final String id;
  final String title;
  final String? subtitle;
  final String status;
  final DateTime createdAt;
  final DateTime updatedAt;
  final int stopsCount;
  final int entriesCount;
  final int photosCount;

  const Trip({
    required this.id,
    required this.title,
    this.subtitle,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
    this.stopsCount = 0,
    this.entriesCount = 0,
    this.photosCount = 0,
  });

  factory Trip.fromJson(Map<String, dynamic> j) => Trip(
        id: j['id'] as String,
        title: j['title'] as String,
        subtitle: j['subtitle'] as String?,
        status: j['status'] as String? ?? 'draft',
        createdAt: DateTime.parse(j['created_at'] as String),
        updatedAt: DateTime.parse(j['updated_at'] as String),
        stopsCount: j['stops_count'] as int? ?? 0,
        entriesCount: j['entries_count'] as int? ?? 0,
        photosCount: j['photos_count'] as int? ?? 0,
      );
}
